query/regexp_test.go

```diff
@@ -65,3 +65,20 @@ func TestLowerRegexp(t *testing.T) {
 		t.Errorf("got mutated original %s want %s", re.String(), in)
 	}
 }
+
+func TestConvertCapture(t *testing.T) {
+	in := "(hello)world"
+	re := mustParseRE(in)
+
+	got, err := ConvertCapture(re)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := mustParseRE("(?:hello)world")
```

Comment:
out of interest what is the value of `LiteralPrefix` in this example? I would hope it is `helloworld`. IE does this simplify down to a simple literal? Maybe our test case needs to have an or group in the capture, this would be the more typical reason we have a capture group.

Reply:
It does simplify down to just the literal `helloworld`.
I added in some new test cases to better illustrate this behaviour as well as demonstrate more typical behaviour